src/main/java/emissary/place/CoordinationPlace.java

```diff
@@ -239,30 +232,108 @@ protected List<IBaseDataObject> coordinate(IBaseDataObject d, boolean hd) {
             }
         }
 
-        if (!errorOccurred) {
-            // Process the ouptut form according to configuration
-            if (outputForm != null) {
-                if (pushForm) {
-                    d.pushCurrentForm(outputForm);
-                } else {
-                    d.setCurrentForm(outputForm);
-                }
-            }
+        applyForm(d, errorOccurred);
 
-            // Clean up my proxies
-            nukeMyProxies(d);
+        sproutHook(d, hd, sproutCollection);
 
-        }
+        // Allow derived classes a shot to clean up the parent
+        cleanUpHook(d);
+
+        return sproutCollection;
+    }
 
+    private void sproutHook(IBaseDataObject d, boolean hd, List<IBaseDataObject> sproutCollection) {
         // Allow derived classes a shot at the sprouts
         if (hd) {
             sproutHook(sproutCollection, d);
         }
+    }
 
-        // Allow derived classes a shot to clean up the parent
-        cleanUpHook(d);
+    /**
+     * Allow derived classes a shot to handle a place exception
+     *
+     * @param p place that was processing when the exception was thrown
+     * @param hd true if doing heavy-duty processing
+     * @param ex exception thrown by the place
+     * 
+     * @return if an error occurred
+     */
+    protected boolean handledPlaceException(IServiceProviderPlace p, boolean hd, Exception ex) {
+        logger.warn("agentProcess{} called from Coordinate problem", (hd ? "HeavyDuty" : "Call"), ex);
+        return true;
+    }
 
-        return sproutCollection;
+    /**
+     * Return false when an error has occurred and the places is configured to not continue
+     *
+     * @param p place that is currently processing the ibdo
+     * @param errorOccurred true if an error occurred
+     *
+     * @return false if processing should not continue
+     */
+    protected boolean shouldContinue(IServiceProviderPlace p, boolean errorOccurred) {
```

Comment:
Let's move this new overload to be adjacent to the existing `shouldContinue` method